Makefile

```diff
@@ -49,7 +52,7 @@ DOCKER_MOUNT := $(if $(BIND_DIR),-v "$(CURDIR)/$(BIND_DIR):/go/src/github.com/do
 # This allows the test suite to be able to run without worrying about the underlying fs used by the container running the daemon (e.g. aufs-on-aufs), so long as the host running the container is running a supported fs.
 # The volume will be cleaned up when the container is removed due to `--rm`.
 # Note that `BIND_DIR` will already be set to `bundles` if `DOCKER_HOST` is not set (see above BIND_DIR line), in such case this will do nothing since `DOCKER_MOUNT` will already be set.
-DOCKER_MOUNT := $(if $(DOCKER_MOUNT),$(DOCKER_MOUNT),-v /go/src/github.com/docker/docker/bundles)
+DOCKER_MOUNT := $(if $(DOCKER_MOUNT),$(DOCKER_MOUNT) -v $(CURDIR)/.git:/go/src/github.com/docker/docker/.git,-v /go/src/github.com/docker/docker/bundles -v $(CURDIR)/.git:/go/src/github.com/docker/docker/.git)
```

Comment:
This line could be written more concisely by taking the second `-v` part completely out of the `$(if...)`, ie:
```make
DOCKER_MOUNT := $(if $(DOCKER_MOUNT),$(DOCKER_MOUNT),-v /go/src/github.com/docker/docker/bundles) -v $(CURDIR)/.git:/go/src/github.com/docker/docker/.git
```

I'm definitely concerned about what this might mean for remote daemons (since `DOCKER_MOUNT` already takes `DOCKER_HOST` into account), but I'm not sure that there's a good solution for that.  Perhaps it'd be fine if those `validate` scripts simply are not supported with a remote daemon unless one explictly removes `.git` from the `.dockerignore`?